stingray/deadtime/model.py

```diff
@@ -0,0 +1,151 @@
+from stingray.utils import njit, prange
+from math import gamma
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy import log
+
+
+def r_in(td, r_0):
+    """Calculate incident countrate given dead time and detected countrate."""
+    tau = 1 / r_0
+    return 1. / (tau - td)
+
+
+def r_det(td, r_i):
+    """Calculate detected countrate given dead time and incident countrate."""
+    tau = 1 / r_i
+    return 1. / (tau + td)
+
+
+@njit()
+def factorial(n):
```

Comment:
Is this faster than the scipy implementation? If so, I think we might want to use this in other parts of the code (e.g. the modelling subpackage) where these functions appear ...

Reply:
It's just a wrapper around the numba-compiled Gamma function. I don't know if it's faster frankly :)

Reply:
Is there an easy way to test this? I don't think I have numba and I've never used numba before, but if it is, I think this would be super useful to have in `utils.py` rather than here, because there are parts of the modelling that heavily rely on factorials and Gamma functions

Reply:
Ok, I made some quick test and scipy's version is faster (tens of microseconds instead of milliseconds) and more reliable (this version spits `inf` with factorial(1000), scipy's gives a result..).

Reply:
Aaaah NOW I remember why. Numba is very picky when it comes to the functions you can call from inside numba functions. It did not accept Scipy's function. 
I changed again. Now I just create a lookup table with the first 150 factorials, calculated with Scipy, at import time, and numba is able to use that. This is still faster, because there are no repeated calculations.